repl/src/main/scala/ammonite/repl/interp/Preprocessor.scala

```diff
@@ -52,8 +52,24 @@ object Preprocessor{
     val TypeDef = DefProc("type"){ case m: G#TypeDef => m.name }
 
     val PatVarDef = Processor { case (name, code, t: G#ValDef) =>
+      //Function to lift lhs expressions into anonymous functions so they will be JITed
+      def wrap(code: String)={
+        import fastparse._
+        import scalaparse.Scala._
+        val par = P( ( `implicit`.? ~ `lazy`.? ~ ( `var` | `val` ) ~! BindPattern.rep1("," ~! Pass) ~ (`:` ~! Type).?).! ~ (`=` ~! StatCtx.Expr.!) )
+        val Result.Success((lhs, rhs), _) = par.parse(code)
+        //Rebuilding definition from parsed data to lift rhs to anon function
+        s"$lhs = { () =>\n $rhs \n}.apply"
+      }
+
       Output(
-        code,
+        //Only wrap rhs in function if it is not a function
+        //Wrapping functions causes type inference errors.
```

Comment:
Can you give an example of a type inference error this will cause if we leave it out? I'm wondering if it's possible to cause problems in cases other than just function and partial function literals